tests/test_stl.cpp

```diff
@@ -177,7 +177,7 @@ TEST_SUBMODULE(stl, m) {
           [](const std::vector<bool> &v) { return v.at(0) == true && v.at(1) == false; });
     // Unnumbered regression (caused by #936): pointers to stl containers aren't castable
     static std::vector<RValueCaster> lvv{2};
-    m.def("cast_ptr_vector", []() { return &lvv; });
+    m.def("cast_ptr_vector", []() { return new decltype(lvv){lvv}; });
```

Comment:
This test no longer serves it's original regression testing purpose with this change: see the  comment above.